lib/init/config-initializer.js

```diff
@@ -50,6 +51,16 @@ function writeFile(config, format) {
         extname = ".json";
     }
 
+    const pkgJSONPath = npmUtils.findPackageJson();
+
+    if (pkgJSONPath) {
+        const pkgJSONContents = JSON.parse(fs.readFileSync(pkgJSONPath, "utf8"));
+
+        if (pkgJSONContents.type === "module") {
+            extname = ".cjs";
+        }
+    }
```

Comment:
I think this would create `.eslintrc.cjs` even if the user has selected YAML or JSON.

Reply:
Oh yes, I will push an update 👍🏻